libr/util/uleb128.c

```diff
@@ -25,13 +26,24 @@ R_API const ut8 *r_uleb128(const ut8 *data, int datalen, ut64 *v) {
 				c = *(data++) & 0xff;
 				if (s > 63) {
 					eprintf ("r_uleb128: undefined behaviour in %d shift on ut32\n", (int)s);
+					if (success) {
+						*success = false;
+					}
+					break;
 				} else {
 					sum |= ((ut64) (c & 0x7f) << s);
 				}
 				if (!(c & 0x80)) {
+					malformed_uleb = false;
 					break;
 				}
 			}
+			if (malformed_uleb) {
+				eprintf ("malformed uleb128\n");
```

Comment:
This `eprintf` should not be here IMO. At most, it can be printed by the callers, but utilities functions should not print anything.

Reply:
I agree. My initial intention was to make error a char ** argument that would store the error message in case of an error, but since there was already an eprintf in r_uleb128,
https://github.com/radareorg/radare2/blob/eab60612287c2d7680c9b4a80e1b7489bcc2dd5a/libr/util/uleb128.c#L27
I decided to comply with the current implementation and eprintf from within r_uleb128.

So, do you suggest I should refactor the code to eprintf from the caller, in both eprintf cases?

Reply:
Yes, just move the eprintf in mach0.c. Don't care about all other callers for now ;)